088-rebalance-progress-status

```diff
@@ -0,0 +1,328 @@
+Firstly, this would help emphasize that a rebalance had not started or had completed by having a percentageComplete: 0% on "ProposalReady" and a percentageComplete: 100% on "Ready". 
+This emphasis could help clear up ambiguity surrounding what the KafkaRebalance “Ready” state or “optimizationResult” field means. 
+Secondly, but more importantly, it would provide an estimate for the minimum time a partition rebalance proposal would take to execute before even executing it. 
+This feature would be of great value to users. 
+However, providing an accurate estimation for this is non-trivial, namely the “estimatedTimeToCompletion” field for “ProposalReady" state, is non-trivial. 
+
+Leveraging the Cruise Control configurations and user-provided network capacity settings, we could provide a rough estimate for “estimatedTimeToCompletetion” field for inter-broker balances. 
+However, one challenge is coming up with a method of reliably determining a reasonable estimate for the disk read/write throughput. 
+It is not so much of an issue for inter-broker rebalance estimates (assuming network is the bottleneck for inter-broker balances) but is certainly an issue for intra-broker rebalance estimates. 
+
+Estimation for inter-broker partition rebalance time:
+```
+# The maximum number of partition movements given CC partition movement cap
+max_partition_movements= min(<# of brokers> *  
+    num.concurrent.partition.movements.per.broker) 
+max_partition_movements=min(max_partition_movements, max.num.cluster.partition.movements)
+
+# The network bandwith given CC bandwith throttle
+bandwidth = min(<network_capacity>, replication.throttle)
+
+
+# The throughput given the max allowed number of partition movements and network bandwidth
+throughput = max_partition_movements * bandwidth
+
+estimatedTimeToCompletion = dataToMoveMB / throughput
+```
+
+However, without an estimate for disk read/write throughput, it is challenging to provide an accurate estimate for intra-broker rebalances but as mentioned above, getting disk throughput is non-trivial for Strimzi. 
+We would either need some estimation of the disk throughput, make it user configurable, or hardcode the value ourselves. 
+
+```
+# The maximum number of partition movements given CC partition movement cap
+max_partition_movements= min(<# of brokers> *      
+    num.concurrent.intra.broker.partition.movements.per.broker)
+max_partition_movements=min(max_partition_movements, max.num.cluster.movements)
+
+estimatedDiskThroughput = ???
+
+# The throughput given the max allowed number of partition movements and disk throughput
+throughput = max_partition_movements * estimatedDiskThroughput
+
+estimatedTimeToCompletion = intraBrokerDataToMoveMB / throughput
+```
+
+Given that its inclusion is not completely necessary and adds significant complexity to the proposal, it is out of scope for this proposal.
```

Comment:
We could just ignore intra-broker movements and just base the estimate on inter-broker movements (they will take up the bulk of the time anyway). We could document it as a theoretical minimum and state that it **will** take longer than this. But it would give a ball park estimate, which is better than the current situation.

Reply:
> We could just ignore intra-broker movements and just base the estimate on inter-broker movements (they will take up the bulk of the time anyway).

Assuming the disk throughput is always faster than the network throughput!

> We could document it as a theoretical minimum and state that it will take longer than this. But it would give a ball park estimate, which is better than the current situation.

Let me think more on this

Reply:
Still looking into this, investigating possible alternatives and hacking it in the prototype to gauge how complicated it would be to implement. If it isn't too complicated,  I add it into this proposal and we can aim for supporting all `KafkaRebalance`  states in one go